management/management.go

```diff
@@ -127,18 +127,19 @@ func New(domain, clientID, clientSecret string, options ...apiOption) (*Manageme
 		option(m)
 	}
 
-	oauth2 := client.OAuth2(m.url, clientID, clientSecret)
+	if m.http == nil {
+		oauth2 := client.OAuth2(m.url, clientID, clientSecret)
 
-	_, err = oauth2.Token(m.ctx)
-	if err != nil {
-		return nil, err
+		_, err = oauth2.Token(m.ctx)
+		if err != nil {
+			return nil, err
+		}
+		m.http = client.New(m.ctx, oauth2)
+		m.http = client.WrapDebug(m.http, m.debug)
+		m.http = client.WrapUserAgent(m.http, m.userAgent)
+		m.http = client.WrapRateLimit(m.http)
```

Comment:
I think we should keep these independent of the client, so we can accept `WithHTTPClient` and `WithUserAgent` options simultaneously for example.

So perhaps these are better left outside the `if m.http == nil { ... }` block

Reply:
I had the problem that `client.WrapUserAgent()` would overwrite a users `http.Client.Transport` and any other properties (eg; `http.Client.Timeout`).

I think I've come up with a way which we can wrap `http.Client` without overwriting bits of the users http.Client. Could you let me know how you feel about https://github.com/go-auth0/auth0/pull/112/files#diff-2c846caea57b2b1d6e49c6417f337625R73-R76?

My tests so far are showing that when `client.WrapUserAgent()` is run outside of `if m.http == nil { ... }` and the user supplies a `http.Client`, that the users client is preserved.

I'm now working on figuring out how I can use Transport with `client.WrapDebug()` and then `client.WrapRateLimit()`.